packages/web3-eth-abi/src/api/functions_api.ts

```diff
@@ -143,3 +143,67 @@ export const encodeFunctionCall = (
 		params ?? [],
 	).replace('0x', '')}`;
 };
+
+/**
+ * Decodes a function call data using its `JSON interface` object.
+ * The JSON interface spec documentation https://docs.soliditylang.org/en/latest/abi-spec.html#json
+ * @param functionsAbi - The `JSON interface` object of the function.
+ * @param data - The data to decode
+ * @param methodSignatureProvided - (Optional) if `false` do not remove the first 4 bytes that would rather contain the function signature.
+ * @returns - The data decoded according to the passed ABI.
+ */
+export const decodeFunctionCall = (
+	functionsAbi: AbiFunctionFragment | AbiConstructorFragment,
+	data: HexString,
+	methodSignatureProvided = true,
+): DecodedParams & { __method__: string } => {
+	const value =
+		methodSignatureProvided && data && data.length >= 10 && data.startsWith('0x')
+			? data.slice(10)
+			: data;
+	if (!functionsAbi.inputs) {
+		throw new Web3ContractError('No inputs found in the ABI');
+	}
+	const result = decodeParameters([...functionsAbi.inputs], value);
+	return {
+		...result,
+		__method__: jsonInterfaceMethodToString(functionsAbi),
```

Comment:
```suggestion
		method: jsonInterfaceMethodToString(functionsAbi),
```
why __method__ is used ?